src/Microsoft.AspNetCore.SignalR.Client/HubConnectionExtensions.cs

```diff
@@ -89,6 +90,11 @@ public static ReadableChannel<TResult> Stream<TResult>(this HubConnection hubCon
             return outputChannel.In;
         }
 
+        private static void On(this HubConnection hubConnetion, string methodName, Type[] parameterTypes, Action<object[]> handler)
```

Comment:
No, keep it private, there's no reason to have the sync overload, they can use the async one.